datafusion/optimizer/Cargo.toml

```diff
@@ -41,6 +41,7 @@ async-trait = { workspace = true }
 chrono = { workspace = true }
 datafusion-common = { workspace = true, default-features = true }
 datafusion-expr = { workspace = true }
+datafusion-functions = { workspace = true }
```

Comment:
I think we need to make some interface for rewriting the function through 🤔 

Perhaps adding a method to `SimplifyInfo` like

```rust
/// return a new Expr with a function call for exact string prefix max
fn make_starts_with(base: Expr, prefix: Expr) -> Option<Expr>
```

Reply:
Hmm not sure what you mean. Is this a pattern already in use somewhere I can learn from?

Reply:
I was suggest adding a method to https://docs.rs/datafusion/latest/datafusion/logical_expr/simplify/trait.SimplifyInfo.html

However i can't quite figure out how to plumb it in without getting messy

THe trick is that this rewrite is based on the semantcs of `starts_with` -- so I think logically it would need to live along side the implementation of `starts_with` because it depends on those semantics.

And similarly, to do the rewrite in the pruning predicate we need something similar (somehow to provide a per function rewrite) 🤔

Reply:
Hmm okay I'm happy to work on this more but I fear I am not familiar enough with the relationship between all of these things to figure out how to wire everything up correctly. Maybe we pause this work until someone with more familiarity of these bits of code can suggest a way to structure this?

Reply:
I will try and help push this forward this week. I am sorry for the delay but I have been super busy the last few days

Reply:
No problem 😄